JsonSchema.OpenApi/DiscriminatorKeyword.cs

```diff
@@ -106,13 +107,13 @@ internal class Model
 	/// <param name="typeToConvert">The type to convert.</param>
 	/// <param name="options">An object that specifies serialization options to use.</param>
 	/// <returns>The converted value.</returns>
+	[UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = "We guarantee that the SerializerOptions covers all the types we need for AOT scenarios.")]
+	[UnconditionalSuppressMessage("AOT", "IL3050:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = "We guarantee that the SerializerOptions covers all the types we need for AOT scenarios.")]
```

Comment:
We should just add `Model` to the context and pass in the type info.

Reply:
It's there, what we are missing is SerializeTo/from JsonNode overload to help statically verify this. It just moves the suppression to a different place though, as I mentioned in the other comment -- there's no way to take an options and combine it with a TypeInfo in the STJ methods so this seems the best we can do for now.